File: 5_software_and_its_engineering/0_software_notations_and_tools/unified_discipline_of_computer_programming/programs_per_computational_model/0_declarative/arrays_and_strings/count_largest_object.py
def birthday_cake_candles(ar: [int]) -> int:
    max_height = 0 if len(ar) == 0 else max(ar)
    return ar.count(max_height)


def main() -> None:
    assert birthday_cake_candles([3, 2, 1, 3]) == 2
    assert birthday_cake_candles([3, 2, 1, 3, 7, 7, 7, 1, 2, 5, 3, 3, 5, 2, 1]) == 3

    assert birthday_cake_candles([3, 2, 1]) == 1

    assert birthday_cake_candles([]) == 0
    assert birthday_cake_candles([1]) == 1
    assert birthday_cake_candles([1, 2]) == 1

    print("completed successfully")


if __name__ == "__main__":
    main()

File: 5_software_and_its_engineering/2_software_creation_and_management/computer_programming/programs_per_computational_model/0_declarative/arrays_and_strings/count_largest_object.scala
def birthday_cake_candles(ar: Array[Int]): Int =
  val length = ar.length
  val max_height = if (length == 0) 0 else ar.reduceLeft(_ max _)
  ar.count(x => x == max_height)

@main
def main(): Unit =
  assert(birthday_cake_candles(Array(3, 2, 1, 3)) == 2)

  assert(
    birthday_cake_candles(
      Array(3, 2, 1, 3, 7, 7, 7, 1, 2, 5, 3, 3, 5, 2, 1)
    ) == 3
  )

  assert(
    birthday_cake_candles(Array(3, 2, 1)) == 1
  )

  assert(birthday_cake_candles(Array()) == 0)
  assert(birthday_cake_candles(Array(1)) == 1)
  assert(birthday_cake_candles(Array(1, 2)) == 1)

  println("completed successfully")
